assets/js/components/KeyMetrics/MetricsSelectionPanel/Footer.js

```diff
@@ -244,25 +244,43 @@ export default function Footer( {
 		setPrevIsOpen( isOpen );
 	}, [ isOpen, prevIsOpen ] );
 
+	let metricsLimitError;
+	if ( selectedMetrics?.length < 2 ) {
+		metricsLimitError = sprintf(
+			/* translators: %d: Number of selected metrics */
+			__( 'Select at least 2 metrics (%d selected)', 'google-site-kit' ),
+			selectedMetrics?.length
+		);
+	} else if ( selectedMetrics?.length > 4 ) {
+		metricsLimitError = sprintf(
+			/* translators: %d: Number of selected metrics */
+			__( 'Select up to 4 metrics (%d selected)', 'google-site-kit' ),
+			selectedMetrics?.length
+		);
+	}
+
 	return (
 		<footer className="googlesitekit-km-selection-panel-footer">
 			{ saveError && <ErrorNotice error={ saveError } /> }
 			<div className="googlesitekit-km-selection-panel-footer__content">
-				{ haveSettingsChanged && selectedMetrics?.length < 2 ? (
+				{ haveSettingsChanged && metricsLimitError ? (
 					<ErrorNotice
 						error={ {
-							message: __(
-								'Select at least 2 metrics',
-								'google-site-kit'
-							),
+							message: metricsLimitError,
 						} }
-						noPrefix={ selectedMetrics?.length < 2 }
+						noPrefix={
+							selectedMetrics?.length < 2 ||
+							selectedMetrics?.length > 4
+						}
 					/>
 				) : (
 					<p className="googlesitekit-km-selection-panel-footer__metric-count">
 						{ sprintf(
 							/* translators: 1: Number of selected metrics, 2: Number of selectable metrics */
-							__( '%1$d of %2$d selected', 'google-site-kit' ),
+							__(
+								'%1$d selected (up to %2$d)',
+								'google-site-kit'
```

Comment:
Since per design (up to %2$d) part is in light grey:
<img width="224" alt="image" src="https://github.com/google/site-kit-wp/assets/10414808/16cb211f-4e9c-4baa-a0c8-b7e44c9f2895">

You can wrap it in `span` and adjust the color via css. Optimal way it can be split and kept as translatable string within span, but since we are already using `span` elements inside translatable string, it seems fine to do so as well.